File: src/main/java/com/iponchallenge/entity/Expense.java
package com.iponchallenge.entity;

import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.CreationTimestamp;

import java.math.BigDecimal;
import java.time.LocalDate;
import java.time.LocalDateTime;
import java.util.UUID;

@Entity
@Table(name = "expenses")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Expense {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id", nullable = false)
    private User user;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "category_id")
    private ExpenseCategory category;

    @Column(nullable = false, precision = 12, scale = 2)
    private BigDecimal amount;

    private String notes;

    @Column(name = "expense_date", nullable = false)
    private LocalDate expenseDate;

    @Enumerated(EnumType.STRING)
    @Column(name = "transaction_type", nullable = false)
    @Builder.Default
    private TransactionType transactionType = TransactionType.EXPENSE;

    @CreationTimestamp
    @Column(name = "created_at", updatable = false)
    private LocalDateTime createdAt;
}

File: frontend/src/types/expense.ts
export type TransactionType = 'EXPENSE' | 'INCOME' | 'GOAL_CONTRIBUTION';
export type CategoryType = 'EXPENSE' | 'INCOME';

export interface ExpenseCategory {
  id: string;
  name: string;
  type: CategoryType;
}

export interface Expense {
  id: string;
  categoryId: string | null;
  categoryName: string;
  amount: number;
  notes: string | null;
  expenseDate: string;
  transactionType: TransactionType;
  createdAt: string;
}

export interface ExpensePayload {
  categoryId?: string;
  amount: number;
  notes?: string;
  expenseDate: string;
  transactionType?: TransactionType;
}
